document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -182,10 +186,28 @@ public static String parseNonCompositeFilterWithCasting(
       String fieldName, String columnName, String op, Object value, Builder paramsBuilder) {
     String parsedExpression =
         prepareCast(prepareFieldDataAccessorExpr(fieldName, columnName), value);
+    if (isInOp(op)) {
+      if (isFirstClassColumn(parsedExpression)) {
+        parsedExpression = "to_jsonb(" + parsedExpression + ")";
+      } else {
+        parsedExpression = parsedExpression.replace("->>", "->");
```

Comment:
Wrapping with `PostgresFieldIdentifierExpressionVisitor` instead of `PostgresDataAccessorIdentifierExpressionVisitor` would directly give you in the format we need without explicit replace.